Update documentation with details about optional argument (#175)
ipdk-io/networking-recipe: docs/clients/p4rt-ctl.rst

For information on security and enabling TLS on gRPC ports, refer to
guides/security-guide.md

> Usage: p4rt-ctl [OPTIONS] COMMAND [ARG...]

Reviewer: The usage text renders badly. I think you need to make it a code block.

(You can visualize it by selecting the **View file** option from the ellipsis menu in github)

Author: Fixed it

Reviewer: Still looks bad.

I think you need to specify a blank line between `code-block` and the display text, and you need to indent each of the lines in the display text:

```text
.. code-block:: text

  Usage: p4rt-ctl [OPTIONS] COMMAND [ARG...]

  positional arguments:
    command               Subcommand to run

  optional arguments:
    -h, --help            show this help message and exit
    -g, --grpc_addr       GRPC_ADDR P4Runtime gRPC server address, format : <server IP>:<port>
```
It looks okay (in VS Code preview) when I do this.

Reviewer: The indentation kind of makes sense. Without the change from indented to unindented, how does the renderer know where the code block ends?

Author: Adding the indentation did the trick. Thanks Derek!

positional arguments:
  command               Subcommand to run

optional arguments:
  -h, --help            show this help message and exit

>   -g GRPC_ADDR, --grpc_addr GRPC_ADDR P4Runtime gRPC server address, format : <server IP>:<port>

Reviewer 2: Is the .rst rendering of document ok for this line? In GitHub, it looks like the line does not have enough whitespace and does not align with previous line

Author: Fixed it and now it's seen as
<img width="360" alt="image" src="https://github.com/ipdk-io/networking-recipe/assets/22621220/d9f62ec7-161e-46f5-a573-53c9340aab22">

> Details about supported option (``-g`` GRPC_ADDR)::

Reviewer: ```suggestion
Details about the supported option (``-g`` GRPC_ADDR)
```
The double colon at the end of the line is displayed in the text. I think this is because you have used in combination with the row of underlines, which makes this line a heading. One does not use a colon on a heading line. The colon serves to set off the following phrase or sentence in normal text. A heading automatically sets off the following text..

---------------------------------------------------

> This is an optional parameter and default value is 127.0.0.1:9559. If P4Runtime server is not running

Reviewer: ```suggestion
This is an optional parameter. The default value is 127.0.0.1:9559. If P4Runtime server is not running
```

> locally, user can specify the address using this option.

Reviewer: ```suggestion
locally, you can specify the address using this option.
```
"user can" is not idiomatic English. It is also impersonal. I strongly recommend writing in the second person ("you"). The sentence structure is simpler and the text is friendlier.

> locally, user can specify the address using this new option.

Reviewer 2: Just use "using this option" instead of "using this new option" (since "new" can be outdated in time and this will stay in documentation)

> Eg: If server is listening on ``5.5.5.5:9559``, command to set pipeline will be:

Reviewer: ```suggestion
For example, if the server is listening on ``5.5.5.5:9559``, the command to set the pipeline will be:
```
_e.g._ is the abbreviation of the Latin expression _exempli gratia_, which means "for instance" or "for example." It should not be used to start a sentence.

 ``p4rt-ctl -g 5.5.5.5:9559 set-pipe sample.pb.bin p4Info.txt``

> Details about supported commands::

Reviewer: Same comment as earlier in the review: end the partial line with a colon or make the line a heading. Don't combine the two.

For information on security and enabling TLS on gRPC ports, refer to
guides/security-guide.md

> .. code-block:: text

Reviewer: That fixes it. :)

Unfortunately, I just noticed the changes to `dpdk-guide.md`. There's a problem...
